tensorflow/core/platform/cloud/gcs_file_system.h

```diff
@@ -142,8 +144,9 @@ class GcsFileSystem : public FileSystem {
   std::unique_ptr<AuthProvider> auth_provider_;
   std::unique_ptr<HttpRequest::Factory> http_request_factory_;
   std::unique_ptr<FileBlockCache> file_block_cache_;
+#ifndef _WIN32
```

Comment:
Is there any particular reason not to include the `GcsDnsCache` on Windows? Empirically we've found it to be important on some cloud configurations, and I think it could affect Windows platforms as well.

/cc @saeta

Reply:
First, the code in class GcsDnsCache is not compilable on Windows, changes are needed. We may turn on it later, in another commit.
Second, I don't know why it is important.  Windows has a builtin DNS cache.  It's also true for Fedora, which runs a local "dnsmasq" for that.  Why do we need an extra cache layer in tensorflow? Is it just for the ones who do not have a local DNS cache?

Reply:
This cache was added to improve the robustness of accessing GCS when running in a container environment (@saeta might be able to say more about the circumstances in which it's useful). I'm inclined to keep the code paths as similar as possible for different platforms, to aid maintainability.

As far as I can tell, the only missing pieces for compiling `GcsDnsCache` on Windows are some changes to the `#include` and a shim to map `inet_ntop()` to the Winsock function `InetNtop()`. If that's the case, I'd ask you to bundle the changes together in the same PR.

Reply:
Thanks for your advice, I'll follow it.